tsdb/db_test.go

```diff
@@ -3803,3 +3803,80 @@ func TestOOOCompaction(t *testing.T) {
 	verifySamples(db.Blocks()[1], 120, 239)
 	verifySamples(db.Blocks()[2], 240, 350) // Merged block.
 }
+
+func TestOOOQuery(t *testing.T) {
+	opts := DefaultOptions()
+	opts.OOOCapMin = 2
+	opts.OOOCapMax = 30
+	opts.OOOAllowance = 24 * time.Hour.Milliseconds()
+	opts.AllowOverlappingBlocks = true
+
+	series1 := labels.FromStrings("foo", "bar1")
+
+	minutes := func(m int64) int64 { return m * time.Minute.Milliseconds() }
+	addSample := func(db *DB, fromMins, toMins int64) {
+		app := db.Appender(context.Background())
+		for min := fromMins; min <= toMins; min += time.Minute.Milliseconds() {
```

Comment:
I think you meant this?
```suggestion
		for min := fromMins*time.Minute.Milliseconds(); min <= toMins; min += time.Minute.Milliseconds() {
```

Reply:
I'm assuming fromMins and toMins are already minutes in milliseconds thats why I increase min by an entire minute in milliseconds on every iteration.

Reply:
Oh right, I missed that you are using `minutes()` everywhere.